react/src/components/atoms/forms/InputCurrency/index.js

```diff
@@ -0,0 +1,241 @@
+                if (e.key === 'ArrowDown') {
+                  newValue = Number.parseFloat(numberValue - step).toFixed(2);
+                  const showError = !validNumber(newValue);
+                  context.updateState({ showError, errorMsg, value: newValue });
+                } else if (e.key === 'ArrowUp') {
+                  newValue = Number.parseFloat(numberValue + step).toFixed(2);
+                  const showError = !validNumber(newValue);
+                  context.updateState({ showError, errorMsg, value: newValue });
+                }
+              }
+            },
+            required: props.required,
+            value: context.value
+          };
+          return<input {...inputAttr} />;
+        }
+      }
+    </InputContext.Consumer>
+  </React.Fragment>
+);
+
+const InputCurrency = (props) => {
+  const {
+    max, min, step, name, onChange, onBlur, placeholder, width, maxlength, format, language, ...inputProps
+  } = props;
+  // Input and Currency share the props.required and props.id values.
+  const currencyProps = {
+    max,
+    min,
+    step,
+    name,
+    placeholder,
+    width,
+    maxlength,
+    required: props.required,
+    id: props.id,
+    onChange,
+    onBlur,
+    format,
+    language
+  };
+  return<Input {...inputProps}><Currency {...currencyProps} /></Input>;
+};
+
+InputCurrency.propTypes = {
+  /** Whether the label should be hidden or not */
+  hiddenLabel: PropTypes.bool,
+  /** The label text for the input field */
+  labelText: PropTypes.string.isRequired,
+  /** Whether the field is required or not */
+  required: PropTypes.bool,
+  /** The unique ID for the input field */
+  id: PropTypes.string.isRequired,
+  /** The name for the input field */
+  name: PropTypes.string.isRequired,
+  /** The max acceptable input length */
+  maxlength: PropTypes.number,
+  /** The pattern to filter input against, e.g. "[0-9]" for numbers only */
+  pattern: PropTypes.string,
+  /** The number of characters wide to make the input field */
+  width: PropTypes.number,
+  /** The placeholder text for the input field */
+  placeholder: PropTypes.string,
+  /** The message to be displayed in the event of an error. */
+  errorMsg: PropTypes.string,
+  /** Custom change function */
+  onChange: PropTypes.func,
+  onBlur: PropTypes.func,
+  /** Default input text value */
+  defaultText: PropTypes.string,
+  /** Max value for the field. */
+  max: PropTypes.number,
+  /** Min value for the field. */
+  min: PropTypes.number,
+  /** Using the up/down arrow keys will increment/decrement the input value by this number. */
+  step: PropTypes.number,
+  /** A language tag that represents what country the currency should display. Comes from IETF BCP 47: https://numbrojs.com/languages.html */
+  language: PropTypes.string,
+  /** Numbro Formatting options for displaying the currency. See https://numbrojs.com/format.html */
+  format: PropTypes.object
+};
+InputCurrency.defaultProps = {
```

Comment:
can you add some more defaults to this:
- step = 0.01 
- language = 'English'
 - format: {
    mantissa: 0,
    trimMantissa: false,
    thousandSeparated: true,
    negative: 'parenthesis'
  }

Reply:
Added.